L1Trigger/L1TZDC/plugins/L1TZDCProducer.cc

```diff
@@ -0,0 +1,4885 @@
+      }  //Old hardcoded lut from @ginnocen - superseded by above
+      else {
+        cEMP = QIE10_regular_fC_full[(UChar_t)(EMP)][idet + 9];
+        cEMM = QIE10_regular_fC_full[(UChar_t)(EMM)][idet];
+      }
+
+      sumcEMP = sumcEMP + cEMP;
+      sumcEMM = sumcEMM + cEMM;
+    }
+    //idet=5-8 correspond to HAD channels
+    for (int idet = 5; idet < 9; idet++) {
+      unsigned short HDP = rawadc[idet + 9][ibx];
+      unsigned short HDM = rawadc[idet][ibx];
+
+      if (!m_doHardCodeLUT) {
+        int cHDP_LUTIndex = zdcLUTIndexHelper(idet + 9, (int)HDP);
+        int cHDM_LUTIndex = zdcLUTIndexHelper(idet, (int)HDM);
+
+        cHDP = ((double)m_params->zdcLUT()->data(cHDP_LUTIndex)) / ((double)m_scaleFactor);
+        cHDM = ((double)m_params->zdcLUT()->data(cHDM_LUTIndex)) / ((double)m_scaleFactor);
+      }  //Old hardcoded lut from @ginnocen - superseded by above
+      else {
+        cHDP = QIE10_regular_fC_full[(UChar_t)(HDP)][idet + 9];
+        cHDM = QIE10_regular_fC_full[(UChar_t)(HDM)][idet];
+      }
+
+      sumcHDP = sumcHDP + cHDP;
+      sumcHDM = sumcHDM + cHDM;
+    }
+    double sumM = sumcEMM + sumcHDM;
+    double sumP = sumcEMP + sumcHDP;
+    //std::cout<<"bx = "<<ibx<<", sumP= "<<sumP<<", sumM= "<<sumM<<std::endl;
+    if (ibx == 4) {
+      std::cout << ", sumM= " << sumM << std::endl;
+      std::cout << ", sumP= " << sumP << std::endl;
+    }
+    l1t::EtSum tempEtM = l1t::EtSum();
+    tempEtM.setHwPt(sumM);
+    tempEtM.setHwEta(-1.);
+    tempEtM.setHwPhi(0.);
+    tempEtM.setType(EtSum::EtSumType::kZDCM);
+
+    l1t::EtSum tempEtP = l1t::EtSum();
+    tempEtP.setHwPt(sumP);
+    tempEtP.setHwEta(1.);
+    tempEtP.setHwPhi(0.);
+    tempEtP.setType(EtSum::EtSumType::kZDCP);
+
+    //One object distinguished by type P, M
+    etsums.push_back(ibx, CaloTools::etSumP4Demux(tempEtP));
+    etsums.push_back(ibx, CaloTools::etSumP4Demux(tempEtM));
+
+    if (ibx >= startBXR && ibx <= endBXR) {
+      etsumsReduced.push_back(ibx - startBXR, CaloTools::etSumP4Demux(tempEtP));
+      etsumsReduced.push_back(ibx - startBXR, CaloTools::etSumP4Demux(tempEtM));
+    }
+  }  // end of loop over bunch crossings
+
+  iEvent.emplace(m_etToken, std::move(etsumsReduced));
+}
+
+// ------------ method called when starting to processes a run  ------------
+void L1TZDCProducer::beginRun(edm::Run const& iRun, edm::EventSetup const& iSetup) {
+  edm::ESHandle<CaloParams> candidateHandle = iSetup.getHandle(m_candidateToken);
+
+  m_params->~CaloParamsHelper();
+  m_params = new (m_params) CaloParamsHelper(*candidateHandle.product());
+  m_scaleFactor = m_params->zdcLUT()->data(0);  //First position is the integer scaling factor
+  std::cout << "SCALE FACTOR FOR LUT: " << m_scaleFactor << std::endl;
+}
+
+// ------------ method called when ending the processing of a run  ------------
+void L1TZDCProducer::endRun(edm::Run const&, edm::EventSetup const&) {}
```

Comment:
If this method is empty it, can it be removed ?

Reply:
It is removed in latest commit